src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/RemoteTagHelperProviderService.cs

```diff
@@ -49,5 +55,12 @@ private async ValueTask<TagHelperResolutionResult> GetTagHelpersCoreAsync(RazorP
             var resolutionResult = await RazorServices.TagHelperResolver.GetTagHelpersAsync(workspaceProject, projectHandle.Configuration, factoryTypeName, cancellationToken).ConfigureAwait(false);
             return resolutionResult;
         }
+        public async ValueTask<TagHelperDeltaResult> GetTagHelpersDeltaCoreAsync(RazorPinnedSolutionInfoWrapper solutionInfo, ProjectSnapshotHandle projectHandle, string factoryTypeName, int lastResultId, CancellationToken cancellationToken)
+        {
+            var tagHelperResolutionResult = await GetTagHelpersCoreAsync(solutionInfo, projectHandle, factoryTypeName, cancellationToken).ConfigureAwait(false);
```

Comment:
We still have to pay the cost of computing all of the TagHelpers for the current solution, this PR only optimizes transmitting that data.